cranelift-filetests/src/test_unwind.rs

```diff
@@ -0,0 +1,233 @@
+
+        write!(
+            text,
+            "Unwind code: offset = {}, operation = {:#?}, info = {}",
+            code.offset(),
+            code.operation(),
+            code.info(),
+        )
+        .unwrap();
+
+        match code.additional_nodes() {
+            0 => {
+                writeln!(text).unwrap();
+                i += 1;
+            }
+            1 => {
+                let offset = offset + core::mem::size_of::<UnwindCode>() as isize;
+                writeln!(text, ", value = {}", unsafe {
+                    core::ptr::read_unaligned(mem.as_ptr().offset(offset) as *const u16)
+                })
+                .unwrap();
+                i += 2;
+            }
+            2 => {
+                let offset = offset + core::mem::size_of::<UnwindCode>() as isize;
+                writeln!(text, ", value = {}", unsafe {
+                    core::ptr::read_unaligned(mem.as_ptr().offset(offset) as *const u32)
+                })
+                .unwrap();
+                i += 3;
+            }
+            _ => panic!("unexpected number of additional nodes"),
+        };
+    }
+}
+
+#[repr(C)]
+struct UnwindInfo {
+    version_and_flags: u8,
+    prologue_size: u8,
+    unwind_code_count: u8,
+    frame_register: u8,
+}
+
+impl UnwindInfo {
+    unsafe fn from_ptr(ptr: *const u8) -> Self {
+        core::ptr::read_unaligned(ptr as *const Self)
+    }
+
+    fn version(&self) -> u8 {
+        self.version_and_flags & 0x3
+    }
+
+    fn flags(&self) -> u8 {
+        (self.version_and_flags & 0xF8) >> 3
+    }
+
+    fn prologue_size(&self) -> u8 {
+        self.prologue_size
+    }
+
+    fn unwind_code_count(&self) -> u8 {
+        self.unwind_code_count
+    }
+
+    fn frame_register(&self) -> u8 {
+        self.frame_register & 0xF
+    }
+
+    fn frame_register_offset(&self) -> u8 {
+        (self.frame_register & 0xF0) >> 4
+    }
+}
+
+#[repr(C)]
+struct UnwindCode {
+    offset: u8,
+    code_and_info: u8,
+}
+
+#[derive(Debug)]
+enum UnwindOperation {
+    PushNonvolatileRegister,
+    LargeStackAlloc,
+    SmallStackAlloc,
+    SetFramePointer,
+    SaveNonvolatileRegister,
+    SaveNonvolatileRegisterFar,
+    SaveXmm128,
+    SaveXmm128Far,
+    PushMachineFrame,
+}
+
+impl From<u8> for UnwindOperation {
+    fn from(value: u8) -> Self {
+        match value {
+            0 => Self::PushNonvolatileRegister,
+            1 => Self::LargeStackAlloc,
+            2 => Self::SmallStackAlloc,
+            3 => Self::SetFramePointer,
+            4 => Self::SaveNonvolatileRegister,
+            5 => Self::SaveNonvolatileRegisterFar,
+            6 => Self::SaveXmm128,
+            7 => Self::SaveXmm128Far,
+            8 => Self::PushMachineFrame,
+            _ => panic!("unsupported unwind operation"),
+        }
+    }
+}
+
+impl UnwindCode {
+    unsafe fn from_ptr(ptr: *const u8) -> Self {
+        core::ptr::read_unaligned(ptr as *const Self)
```

Comment:
Please take a slice and read the individual fields instead.

Reply:
Agreed, that would be better.  I'll fix.